crates/forge/src/lib.rs

```diff
@@ -101,16 +102,19 @@ fn collect_tests_from_directory(
 fn find_cairo_root_files_in_directory(
     package_path: &Utf8PathBuf,
     lib_path: &Utf8PathBuf,
+    subpackages: &[Utf8PathBuf],
 ) -> Result<Vec<Utf8PathBuf>> {
     let mut test_files: Vec<Utf8PathBuf> = vec![lib_path.clone()];
     let src_path = lib_path.parent().with_context(|| {
         format!("Failed to get parent directory of a package at path = {lib_path}")
     })?;
-
     for entry in WalkDir::new(package_path)
         .sort_by_file_name()
         .into_iter()
-        .filter_entry(|e| !e.path().starts_with(src_path))
+        .filter_entry(|e| {
+            !e.path().starts_with(src_path)
+                && !subpackages.iter().any(|path| e.path().starts_with(path))
+        })
```

Comment:
Yup, that's the reason + we avoid collecting tests from unwanted packages. It is really only useful when in the root package of the workspace root. But actually now that I think about it I can ditch it entirely and just implement #654 here, it will help us get rid of this weird logic, wdyt?

Reply:
Yup, did it like I described and it work without this weird subpackages logic